spark/v3.1/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteMergeInto.scala

```diff
@@ -228,13 +225,6 @@ case class RewriteMergeInto(spark: SparkSession) extends Rule[LogicalPlan] with
     }
   }
 
-  private def isCardinalityCheckEnabled(table: Table): Boolean = {
-    PropertyUtil.propertyAsBoolean(
-      table.properties(),
-      MERGE_CARDINALITY_CHECK_ENABLED,
```

Comment:
Okay, it looks like this property is still used by Spark 3.1, so we shouldn't actually remove it (or the ability to disable it!) until we remove support for 3.1.

Reply:
I've restored this and updated the comment on the property itself to mention that we will drop it once Spark 3.1 support will be dropped